libs/components/indicators/testing/src/icon/icon-harness.ts

```diff
@@ -0,0 +1,80 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+
+import { SkyIconHarnessFilters } from './icon-harness-filters';
+
+/**
+ * Harness for interacting with an icon component in tests.
+ */
+export class SkyIconHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-icon';
+
+  #getIcon = this.locatorFor('.sky-icon');
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyIconHarness` that meets certain criteria.
+   */
+  public static with(
+    filters: SkyIconHarnessFilters
+  ): HarnessPredicate<SkyIconHarness> {
+    return SkyIconHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /** Gets the icon name */
+  public async getIconName(): Promise<string | undefined> {
+    const iconClass = await (await this.#getIcon()).getAttribute('class');
+    const regex = /^sky-i-|fa-(?!fw|2xs|xs|sm|lg|2xl|[0-9]+x)/;
+    let className = iconClass
+      ?.split(' ')
+      .find((classValue: string) => classValue.match(regex));
+
+    className = className?.replace(/-line|-solid$/, '');
+    className = className?.replace(/^sky-i-|fa-/, '');
+    return className;
+  }
+
+  /** Gets if the icon has fixed width */
+  public async isFixedWidth(): Promise<boolean> {
+    const icon = await this.#getIcon();
+    return icon.hasClass(`fa-fw`);
+  }
+
+  /** Gets the icon type */
+  public async getIconType(): Promise<string> {
+    const iconClass = await (await this.#getIcon()).getAttribute('class');
```

Comment:
Did you try switching to `getProperty('classList')` instead of splitting the attribute on a space? that will simplify the code a bit.

Reply:
i tried that last week and wasn't able to get half of the functions at this link https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/entries to work locally, but i just thought of a way to do it and its working so brb gonna go implement that change